pkg/beacon/analyzer.go

```diff
@@ -186,20 +186,30 @@ func (a *analyzer) start() {
 				}
 
 				// connection count scoring
-				tsConnDiv := (float64(a.tsMax) - float64(a.tsMin)) / 10.0
+				tsConnDiv := (float64(a.tsMax) - float64(a.tsMin)) / 3600
 				tsConnCountScore := float64(res.ConnectionCount) / tsConnDiv
 				if tsConnCountScore > 1.0 {
 					tsConnCountScore = 1.0
 				}
 
-				//score numerators
-				tsSum := tsSkewScore + tsMadmScore + tsConnCountScore
-				dsSum := dsSkewScore + dsMadmScore + dsSmallnessScore
+				// calculate final ts and ds scores
+				tsScore := math.Ceil(((tsSkewScore+tsMadmScore+tsConnCountScore)/3.0)*1000) / 1000
+				dsScore := math.Ceil(((dsSkewScore+dsMadmScore+dsSmallnessScore)/3.0)*1000) / 1000
 
-				//score averages
-				tsScore := math.Ceil((tsSum/3.0)*1000) / 1000
-				dsScore := math.Ceil((dsSum/3.0)*1000) / 1000
-				score := math.Ceil(((tsSum+dsSum)/6.0)*1000) / 1000
+				// calculate duration score
+				duration := math.Ceil((float64(res.TsList[tsLength]-res.TsList[0])/(float64(a.tsMax)-float64(a.tsMin)))*1000) / 1000
+				if duration > 1.0 {
+					duration = 1.0
+				}
+
+				// calculate histogram score
+				bucketList, freqList, freqCount, histScore := getTsHistogramScore(a.tsMin, a.tsMax, res.TsList)
```

Comment:
Fix #4: Add a score that is dependent on the rate of connections across a number of timestamp buckets.